### PR TITLE
image random crop edge case

### PR DESCRIPTION
**Pull request by autoih**
This PR adds the edge case when size > value.shape by adopting `zero padding` to have the same shape with size.

### CODE REVIEW

**Comment by autoih** [comment]
Hi @BryanCutler, could you please help me review the PR? Thanks!

File: tensorflow/python/ops/random_ops.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 from __future__ import print_function
 
 import numpy as np
+import tensorflow as tf
```
Comment (BryanCutler) [inline tensorflow/python/ops/random_ops.py]: You should follow the convention to only import what you are using. I don't think you should import the top level tensorflow module here

File: tensorflow/python/ops/random_ops.py [hunk 2]
```diff
@@ -313,18 +314,30 @@ def random_crop(value, size, seed=None, name=None):
...
-    limit = shape - size + 1
-    offset = random_uniform(
-        array_ops.shape(shape),
-        dtype=size.dtype,
-        maxval=size.dtype.max,
-        seed=seed) % limit
-    return array_ops.slice(value, offset, size, name=name)
+
+    with tf.Session():
+        size_dim = size.eval()
```
Comment (BryanCutler) [inline tensorflow/python/ops/random_ops.py]: This has a cost to evaluate, I'm not sure its a good idea to do it here

**Review by BryanCutler — COMMENTED** [review]
@autoih can you add a test for this?

File: tensorflow/python/ops/random_ops.py (hunk 2 above)
Comment (autoih) [reply tensorflow/python/ops/random_ops.py]: I see. Thanks, @Bryancutler

File: tensorflow/python/ops/random_ops.py (hunk 1 above)
Comment (autoih) [reply tensorflow/python/ops/random_ops.py]: Thanks @BryanCutler.

**Pull request closed without merging**